x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/physical/local/EnableSpatialDistancePushdown.java

```diff
@@ -94,46 +95,82 @@ private FilterExec rewrite(FilterExec filterExec, EsQueryExec esQueryExec) {
         return filterExec;
     }
 
-    private Map<NameId, StDistance> getPushableDistances(List<Alias> aliases) {
-        LinkedHashMap<NameId, StDistance> refs = new LinkedHashMap<>();
+    private void getPushableDistances(List<Alias> aliases, Map<NameId, StDistance> distances, Map<NameId, Alias> others) {
         aliases.forEach(alias -> {
-            if (alias.child() instanceof StDistance distance && canPushSpatialFunctionToSource(distance, Map.of())) {
-                refs.put(alias.id(), distance);
-            } else if (alias.child() instanceof ReferenceAttribute ref && refs.containsKey(ref.id())) {
-                StDistance distance = refs.get(ref.id());
-                refs.put(alias.id(), distance);
+            if (alias.child() instanceof StDistance distance && canPushSpatialFunctionToSource(distance)) {
+                distances.put(alias.id(), distance);
```

Comment:
The standard way is to have an `AttributeMap` and then `distances.put(alias.toAttribute(), distance)` rather than using the name id directly.

Let's refactor this - if needed in a direct follow-up - since I think it's important to have just 1 way to do things in the optimizer for simplicity's sake.